src/qibocal/auto/execute.py

```diff
@@ -121,23 +121,25 @@ def current(self):
         assert self.head is not None
         return self.graph.task(self.head)
 
-    def run(self):
+    def run(self, mode):
         """Actual execution.
 
         The platform's update method is called if:
         - self.update is True and task.update is None
         - task.update is True
         """
         self.head = self.graph.start
-
         while self.head is not None:
             task = self.current
-            task_execution = task.run(platform=self.platform, qubits=self.qubits)
             completed = Completed(task, Normal(), self.output)
-            completed.data = next(task_execution)
-            completed.results = next(task_execution)
+            task_execution = task.run(platform=self.platform, qubits=self.qubits)
+            if mode.name not in ["report"]:
```

Comment:
```suggestion
            if mode.name != "report":
```
Prefer the simplest operation.